Issue 378: Use lintr::lint_package() and resolve all issues found package wide

seabbs opened the pull request:
<!-- Thanks for opening this pull request! Below we have provided a suggested template for PRs to this repository and a checklist to complete before opening a PR -->
 
## Description

This PR closes #378.

This PR just makes changes in order to get `lintr::lint_package()` to pass without any listing issues. I have opened a new issue to help us prevent this from happening again (#383). This is a blocker for #376 as well as future PRs if we wish their CI to pass without introducing spurious changes.

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/epinowcast/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [x] I have added a news item linked to this PR.
- [x] I have updated the package development version by one increment in both `NEWS.md` and the `DESCRIPTION`.
- [x] I have reviewed CI checks for this PR and addressed them as far as I am able.

<!-- Thanks again for this PR - @epinowcast dev team -->

pearsonca commented on R/check.R:
```diff
@@ -309,13 +309,13 @@ check_numeric_timestep <- function(dates, date_var, timestep, exact = TRUE) {
-  if (!check) {
+  if (check) {
+    return(invisible(NULL))
+  }else {
```
```suggestion
  } else {
```

pearsonca replied:
the linter must complain about that, yeah?

seabbs replied:
Oddly no it doesn't but I have fixed.

pearsonca commented on R/epinowcast.R:
```diff
@@ -162,12 +162,12 @@ epinowcast <- function(data,
 
-  if (!missing$formula %in% "~0") {
+  if (!missing$formula == "~0") {
     warning("The missing data model is highly experimental. There is a
      significant likelihood that bugs are present in its implementation.")
   }
 
-  if (!missing$formula %in% "~0") {
+  if (!missing$formula == "~0") {
     warning("The missing data model is highly experimental. There is a
      significant likelihood that bugs are present in its implementation.")
   }
```
this seems oddly repetitive - am I missing something?

seabbs replied:
Yes, this is a duplication and no I don't know why it is in there!

pearsonca reviewed (commented):
Looks good overall. One space I'm surprised linter didn't catch, one odd bit of code, but maybe I'm miss reading something.

Bisaloo commented on R/data-converters.R:
```diff
@@ -348,13 +348,13 @@ enw_incidence_to_cumulative <- function(obs, by = NULL) {
   min_reference_date = min(obs$reference_date, na.rm = TRUE), copy = TRUE
 ) {
-  stopifnot("The data already has a timestep of a day" = !timestep %in% "day")
+  stopifnot("The data already has a timestep of a day" = !timestep == "day")
```
```suggestion
  stopifnot("The data already has a timestep of a day" = timestep != "day")
```

There is a linter coming for this some it might be worth getting ahead of it here?

Bisaloo commented on R/epinowcast.R:
```diff
@@ -162,12 +162,12 @@ epinowcast <- function(data,
   )
 
-  if (!missing$formula %in% "~0") {
+  if (!missing$formula == "~0") {
```
```suggestion
  if (missing$formula != "~0") {
```

Bisaloo commented on R/epinowcast.R:
```diff
@@ -162,12 +162,12 @@ epinowcast <- function(data,
   }
 
-  if (!missing$formula %in% "~0") {
+  if (!missing$formula == "~0") {
```
```suggestion
  if (missing$formula != "~0") {
```

Bisaloo commented on R/model-modules.R:
```diff
@@ -251,7 +251,7 @@ enw_report <- function(non_parametric = ~0, structural = ~0, data) {
   data_list$rep_t <- data$time[[1]] + data$max_delay[[1]] - 1
   data_list$model_rep <- as.numeric(
-    !as_string_formula(non_parametric) %in% "~1"
+    !as_string_formula(non_parametric) == "~1"
```
```suggestion
    as_string_formula(non_parametric) != "~1"
```

Bisaloo commented on R/model-modules.R:
```diff
@@ -512,12 +512,12 @@ enw_expectation <- function(r = ~ 0 + (1 | day:.group), generation_time = 1,
 enw_missing <- function(formula = ~1, data) {
   if (nrow(data$missing_reference[[1]]) == 0 &&
-    !(as_string_formula(formula) %in% "~0")) {
+    !(as_string_formula(formula) == "~0")) {
```
```suggestion
    as_string_formula(formula) != "~0") {
```

Bisaloo reviewed (commented):
I've made suggestions for this but a better way to write `!x %in% "a"` is `x != "a" rather than `!x == "a"`.

The next lintr version will have a lintr for this so it's probably worth using the better version already?

seabbs commented:
Thanks for the good reviews both. I've added you both to the news item (I also cheekily updated the overall contributors list as it had got a little out of date and added a space).

> but a better way to write

Thanks for this @Bisaloo (me being lazy here). I have no idea if your suggestion will work with switch - I guess we will see.

@pearsonca its not clear to me why `lintr` didn't flag the lack of space ahead of `else` but I've updated.
